pkg/controller/executors/kube.go

```diff
@@ -41,14 +44,34 @@ func (c fallbackClientReader) List(ctx context.Context, list runtime.Object, opt
 	return
 }
 
+type FallbackClientBuilder struct {
+	uncached []client.Object
+}
+
+func (f *FallbackClientBuilder) WithUncached(objs ...client.Object) cluster.ClientBuilder {
+	f.uncached = append(f.uncached, objs...)
+	return f
+}
+
+func (f FallbackClientBuilder) Build(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
+	c, err := client.New(config, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return client.NewDelegatingClient(client.NewDelegatingClientInput{
+		Client: c,
+		CacheReader: fallbackClientReader{
+			orderedClients: []client.Reader{cache, c},
+		},
+		UncachedObjects: f.uncached,
+		// TODO figure out if this should be true?
+		// CacheUnstructured: true,
+	})
+}
+
 // Creates a new k8s client that uses the cached client for reads and falls back to making API
 // calls if it failed. Write calls will always go to raw client directly.
-func NewFallbackClient(cachedClient, rawClient client.Client) client.Client {
-	return client.DelegatingClient{
-		Reader: fallbackClientReader{
-			orderedClients: []client.Client{cachedClient, rawClient},
-		},
-		StatusClient: rawClient,
-		Writer:       rawClient,
-	}
+func NewFallbackClientBuilder() *FallbackClientBuilder {
```

Comment:
I think we could make `FallbackClientBuilder` private and return here `cluster.ClientBuilder`.

The main reason being that we are not adding yet additional attrs to it and we lose the type after `WithUncached` calls.

Reply:
As a general rule of thumb in go (unlike java), one should ["accept interfaces and return structs"](https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a). This allows the receiver to accept the return in whatever interface they define..